R/filter_ard_hierarchical.R

```diff
@@ -125,6 +132,15 @@ filter_ard_hierarchical <- function(x, filter, keep_empty = FALSE) {
     )
   }
   ard_args <- attributes(x)$args
+  by <- ard_args$by
+
+  # get name of filtering variable
+  process_selectors(
```

Comment:
After the selection, do we need a check that they have only included one variable, e.g. `check_scalar(message = "custom message")`